### PR TITLE
Feature/toggle location2

### PR DESCRIPTION
**Pull request by Kapestian**
# Toggle location update 2
note: code was started last week and was then discontinued. This is the same feature but started from a fresh main
## What's new
This PR links the toggle button from the settings screen to the `locationViewmodel` which allows to toggle location update. For now, disabling the location means that other user can still see your profile if they stand at your last known position (where you toggled the switch off). In later updates, the `ProfileViewmodel` will be extended to allow the toggle button to make the user truly invisible to others.
## Codebase changes:
* `Settings.kt`: linked to `LocationViewModel`
* `AroundYou.kt`: Added different empty screens when location update is disabled or permission is not aquired
* `MainActivity.kt`: update components signatures to allow use of `PermissionManager`, `AppDataStore` and `LocationViewModel`

### Special thanks to @arthur-mrgt for his very clean interface for location and permissions

### CODE REVIEW

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/AroundYou.kt [hunk 1]
```diff
@@ -125,25 +138,28 @@ fun AroundYouScreen(
...
+              if (!isConnected.value) {
+                EmptyProfilePrompt(label = "No Internet Connection", testTag = "noConnectionPrompt")
+              } else if (!isDiscoverable) {
+                EmptyProfilePrompt(
+                    label = "Activate location sharing in the app settings!",
+                    testTag = "activateLocationPrompt")
+              } else if (!isTestMode && !hasLocationPermission) {
+                EmptyProfilePrompt(
+                    label =
+                        "Precise location permission is required to show profiles. Activate it in your phone settings",
```
Comment (louiscrc) [inline app/src/main/java/com/github/se/icebreakrr/ui/sections/AroundYou.kt:152]: Nitpick : put this in strings.xml

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/AroundYou.kt [hunk 2]
```diff
@@ -125,25 +138,28 @@ fun AroundYouScreen(
...
-                                  fontSize = TEXT_SIZE_LARGE,
-                                  fontWeight = FontWeight.Bold,
-                                  color = NO_CONNECTION_TEXT_COLOR)
-                            }
-                      }
-                    } else if (filteredProfiles.value.isNotEmpty()) {
+            modifier = Modifier.padding(innerPadding),
+            content = {
+              if (!isConnected.value) {
+                EmptyProfilePrompt(label = "No Internet Connection", testTag = "noConnectionPrompt")
```
Comment (louiscrc) [inline app/src/main/java/com/github/se/icebreakrr/ui/sections/AroundYou.kt:144]: Nitpick : put this in strings.xml

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/AroundYou.kt [hunk 3]
```diff
@@ -125,25 +138,28 @@ fun AroundYouScreen(
...
+                    label = "Activate location sharing in the app settings!",
+                    testTag = "activateLocationPrompt")
+              } else if (!isTestMode && !hasLocationPermission) {
+                EmptyProfilePrompt(
+                    label =
+                        "Precise location permission is required to show profiles. Activate it in your phone settings",
+                    testTag = "noLocationPermissionPrompt")
+              } else if (filteredProfiles.value.isEmpty()) {
+                EmptyProfilePrompt(
+                    label = "There is no one around. Try moving!", testTag = "emptyProfilePrompt")
```
Comment (louiscrc) [inline app/src/main/java/com/github/se/icebreakrr/ui/sections/AroundYou.kt:156]: Nitpick : put this in strings.xml

**Review by louiscrc — APPROVED** [review]
Nice PR Valentin!
I can't make many comments because it's very clean...
Few nitpicks can be corrected, especially with redundant strings that can be added or are already in the strings.xml

**Review by louiscrc — APPROVED** [review]
Looks good to me !
Sorry for the pain of having to merge everyone's PR into yours...

**Review by arthur-mrgt — APPROVED** [review]
Lgtm

**Pull request merged**